HACKING.md

```diff
@@ -0,0 +1,113 @@
+)
+which comes with a Solr schema (MBSSSSS -
+[repository](https://github.com/metabrainz/mbsssss/),
+[tickets](https://tickets.metabrainz.org/issues/?jql=project%20%3D%20SEARCH%20AND%20component%20%3D%20Schema%20AND%20status%20!%3D%20Closed)
+).
+These versions of MBS, SIR and MB Solr should
+(preferably for development, necessarily for production)
+conform to the same version of Relax NG schema (MMD -
+[repository](https://github.com/metabrainz/mmd-schema)
+) as it may just not work together otherwise.
+
+See also [MusicBrainz search architecture](https://musicbrainz.org/doc/Development/Search_Architecture) for a broader overview.
+
+Solr logs
+---------
+
+Logs can be useful when making a search query with MBS or when indexing documents with SIR.
+
+In a terminal, follow Solr logs using:
+
+```sh
+docker-compose logs -f --tail 1 search
+```
+
+For example browse your local MusicBrainz Server instance at
+<http://localhost:5000/search?query=%28Downward+Spiral%29+AND+format%3ACD&type=release&limit=25&method=advanced>.
+Then the request appears as follows in Solr logs.
+```
+[   x:release] o.a.s.c.S.Request [release]  webapp=/solr path=/advanced params={q=(Downward+Spiral)+AND+format:CD&start=0&rows=25&wt=mbjson}
+```
+
+These values can then be used for debugging in Solr Admin query screen; See the next section.
+
+Solr Admin
+----------
+
+To debug how other components (MBS, SIR) interact with the search server,
+Solr Admin web interface (browsable from <http://localhost:8983>) is your friend.
+
+See “[Using the Solr Administration User Interface](https://solr.apache.org/guide/7_7/using-the-solr-administration-user-interface.html)”
+in Apache Solr Reference Guide 7.7 for more information.
+
+### Query screen
+
+The query screen allows to reproduce and tweak search queries made through MBS.
+
+To follow the same example as in the above section “[Solr logs](#solr-logs)”,
+switch the Core Selector to “`release`” and click on “Query” to get to
+<http://localhost:8983/solr/#/release/query>.
+
+Then start filling the form as follows:
+* Set the first field “Request-Handler (qt)” (qt)
+  to the value for `path` in the log record, with a leading slash:
+  `/advanced`
+* Set the second field “q” (query string)
+  to the text you typed for the search
+  (that is the unescaped value for`q` in the log record):
+  `(Downward Spiral) AND format:CD`
+* Select `xml` in the field `wt` (writer type) for the most readable presentation.
+
+From here, different combinations can be choosed depending on your needs:
+
+* Check `debugQuery` to have a detailed score computation (using search fields and boosts) for each result.
+
+* Set `fl` (field list) to
+  either `mbid,score` for the least detailed list of results
+  or `*,score` for the most detailed list of results
+
+* Restrict to a single result by setting “Raw Query Parameters” to:
```

Comment:
replace with:

If you expect another item to show up in the search results but it doesn't, you can ask solr to return the score logging for that item and compare it with the scores of the search results.

Reply:
Applied in commit 458125c